sdk/communication/Azure.Communication.Calling.Server/src/PartitionedDownloader.cs

```diff
@@ -0,0 +1,329 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Communication.Calling.Server.Models;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Calling.Server
+{
+    internal class PartitionedDownloader
+    {
+        /// <summary>
+        /// The client used to download the blob.
+        /// </summary>
+        private readonly ConversationClient _client;
+
+        /// <summary>
+        /// The maximum number of simultaneous workers.
+        /// </summary>
+        private readonly int _maxWorkerCount;
+
+        /// <summary>
+        /// The size of the first range requested (which can be larger than the
+        /// other ranges).
+        /// </summary>
+        private readonly long _initialRangeSize;
+
+        /// <summary>
+        /// The size of subsequent ranges.
+        /// </summary>
+        private readonly long _rangeSize;
+
+        internal PartitionedDownloader(
+            ConversationClient client,
+            ContentTransferOptions transferOptions = default)
+        {
+            _client = client;
+
+            // Set _maxWorkerCount
+            if (transferOptions.MaximumConcurrency.HasValue
```

Comment:
is not it better if `transferOptions.MaximumConcurrency` is not nullable and instead has a default value if not set?

Reply:
I can do the change and solve both related comments with that.